azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/DeploymentTemplateBuilder.java

```diff
@@ -323,6 +326,43 @@ private void addNetworkSecurityGroup() {
 
    }
 
+   // Helper function for simplifying an array of ports to a list of ranges FirewallOptions expects.
+   public static List<String> simplifyPorts(int[] ports){
```

Comment:
Add the corresponding unit tests that verify this method works for the different types of input ports: isolated ones, contiguous, etc.